File: 프로그래머스/2/12949. 행렬의 곱셈/행렬의 곱셈.java
class Solution {
    public int[][] solution(int[][] arr1, int[][] arr2) {
        int[][] answer = new int[arr1.length][arr2[0].length];
        
        for (int j = 0; j < arr1.length; j++) {
            for (int i = 0; i < arr2[0].length; i++) {
                for (int k = 0; k < arr1[0].length; k++) {
                    answer[j][i] += arr1[j][k] * arr2[k][i];
                }
            }
        }
        return answer;
    }
}

File: 프로그래머스/2/12949. 행렬의 곱셈/행렬의 곱셈.py
def solution(arr1, arr2):
    answer = [[0] * len(arr2[0]) for _ in range(len(arr1))] 
    
    for j in range(len(arr1)):
        for i in range(len(arr2[0])):
            for k in range(len(arr1[0])):
                answer[j][i] += arr1[j][k] * arr2[k][i]
    
    
    return answer


# 1 4     3 4 3
# 3 2     5 6 3
# 4 1